target/linux/mediatek/filogic/base-files/lib/upgrade/platform.sh

```diff
@@ -1,5 +1,16 @@
 REQUIRE_IMAGE_METADATA=1
 
+asus_initial_setup()
+{
+	# initialize UBI if it's running on initramfs
+	[ "$(rootfs_type)" = "tmpfs" ] || return 0
+
+	ubirmvol /dev/ubi0 -N rootfs
+	ubirmvol /dev/ubi0 -N rootfs_data
+	ubirmvol /dev/ubi0 -N jffs2
+	ubimkvol /dev/ubi0 -N jffs2 -s 0x3e000
```

Comment:
Why create a jffs2 volume?

Reply:
Oh, I got it. I would suggest check some partition label / size or magic number instead of the `$(rootfs_type)` to identify whether it is an official image. If my understanding is correct each time running initramfs will format the user configs. This is unacceptable because someone like to use initramfs to test new features.

Reply:
or just add the commands to the installation guide...

Reply:
> If my understanding is correct each time **running initramfs** will format the user configs. This is unacceptable because someone like to use initramfs to test new features.

And you search through the entire repository and understand that this is exactly how it is done everywhere.
1) https://github.com/openwrt/openwrt/blob/62277c485e79f6fe453a8a1bd54f271ae17c7490/target/linux/mediatek/filogic/base-files/lib/upgrade/platform.sh#L20
2) https://github.com/openwrt/openwrt/blob/62277c485e79f6fe453a8a1bd54f271ae17c7490/target/linux/qualcommax/ipq807x/base-files/lib/upgrade/buffalo.sh#L22
3) https://github.com/openwrt/openwrt/blob/62277c485e79f6fe453a8a1bd54f271ae17c7490/target/linux/ipq806x/base-files/lib/upgrade/buffalo.sh#L42
4) etc...

You can launch the `initramfs` image in two different ways:
1) through special bootloader commands, copying the image directly into memory
2) use `sysupgrade` to write the image to the flash chip

The `asus_initial_setup` function will only be written in the second case.

Reply:
> Oh, I got it. I would suggest check some partition label / size or magic number instead of the `$(rootfs_type)` to identify whether it is an official image. If my understanding is correct each time running initramfs will format the user configs. This is unacceptable because someone like to use initramfs to test new features.

@DragonBluep  I fully agree with you. 
@remittor We need to implement it in other manner for example as a separate package or script which can be executed manually then tester will not lose data.

Reply:
@patrykk , @DragonBluep 
Why is partition `overlay` boldly deleted on all Xiaomi devices without asking questions, but for Asus routers you need to ask the user?

Reply:
There is no special reason. Just because I didn't review the Xiaomi PR. I'm not a repo maintainer. And it's fine for me if only case 3 trigger the script.

> use sysupgrade to write the tar image to the flash chip (on initramfs mode)